using-mobile-extensions/mobile-core/platform-services.md

```diff
@@ -237,24 +248,53 @@ This is just an implementation example. For more information about handling netw
 **Swift**
 
 ```swift
-class SamplePerformerOverrider: ACPHttpConnectionPerformer {
+import Foundation
+import ACPCore
+
+class SamplePerformerOverrider: NSObject, ACPHttpConnectionPerformer {
+    
+    // Modifications here would allow for conditional overriding based on the url/method.
+    // In this example it always returns true to override all network requests.
     func shouldOverride(_ url: URL, method: String) -> Bool {
-        return true
+      return true
     }
 
+    // Network request override with a completion block. The provided parameters should be configured on the network request.
     func request(_ url: URL, httpCommand command: String, connectPayload payload: String, requestPropertyDict requestProperty: [String : String], connectTimeout: TimeInterval, readTimeout: TimeInterval, completion: @escaping (ACPHttpConnection?) -> Void) {
-        //...
-        let session = URLSession()
-        _ = session.dataTask(with: url) { (data, response, error) in
-            let httpResponse = response as! HTTPURLResponse
-            let connectionOverride = ACPHttpConnection(response: httpResponse, data: data)
-            if let error = error {
-                // Handle error
-                completion(nil)
-            } else {
-                completion(connectionOverride)
-            }
+
+      // Create the URLSessionConfiguration with the provided timeouts
+      let config = URLSessionConfiguration.ephemeral
+      config.urlCache = nil
+      config.timeoutIntervalForRequest = readTimeout
+      config.timeoutIntervalForResource = connectTimeout
+
+      let session = URLSession(configuration: config)
+
+      // Create an NSURLRequest with the provided request parameters
+      let request = NSMutableURLRequest.init(url: url)
+      request.cachePolicy = NSURLRequest.CachePolicy.reloadIgnoringLocalCacheData
+      request.httpMethod = command
+
+      if !payload.isEmpty && "POST" == command.uppercased() {
+        request.httpBody = payload.data(using: .utf8)
+      }
+
+      for property in requestProperty {
+        request.setValue(property.value, forHTTPHeaderField: property.key)
+      }
+
+      // Start the request
+      let task = session.dataTask(with: url, completionHandler: { (data, response, error) in
+        if error == nil {
+          let httpResponse = response as? HTTPURLResponse
+          // create ACPHttpConnection object with the data received and call the completion handler
+          let connectionOverride = ACPHttpConnection(response: httpResponse, data: data)
+                                                                 completion(connectionOverride)
```

Comment:
Is it only on web that the completion(connectionOverride) is tabbed all the way to the right?

Reply:
fixed